### PR TITLE
Device fixes

### PR DESCRIPTION
**Pull request by fbuihuu**

### CODE REVIEW

**Comment by fbuihuu** [comment]
@zonque: I've added a commit log.

**Comment by poettering** [comment]
Please drop the first patch, it looks wrong to me. The second one looks good!

**Comment by fbuihuu** [comment]
@poettering first patch dropped.

File: src/core/device.c [hunk 1]
```diff
@@ -315,12 +315,19 @@ static int device_setup_unit(Manager *m, struct udev_device *dev, const char *pa
...
+        /* The device unit can still be present even if the device was
+         * unplugged: a mount unit can reference it hence preventing
+         * the GC to have garbaged it. That's desired since the device
+         * unit may have a dependency on the mount unit which was
+         * added during the loading of the later. */
+        if (u && DEVICE(u)->state == DEVICE_PLUGGED) {
+                /* This unit is in plugged state: we're sure it's
+                 * attached to a device. */
+                if (!path_equal(DEVICE(u)->sysfs, sysfs)) {
+                        log_unit_error(u, "Dev %s appeared twice with different sysfs paths %s and %s",
```
Comment (janzizazizka) [inline src/core/device.c:327]: Why the log level was changed from debug to error? Was there some particular reason for that? This message seems to be quite often triggered when for example a network interface is renamed. We used to ignore the message but now some of our automated tests fail. Problem with network interface renaming is another issues, however.

**Pull request merged**